cmake/Modules/FindMKLDNN.cmake

```diff
@@ -21,20 +21,24 @@ IF(NOT MKLDNN_FOUND)
 
   if(USE_XPU) # Build oneDNN GPU library
     if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-      set(DNNL_HOST_COMPILER "g++")
+      # Linux
       # g++ is soft linked to /usr/bin/cxx, oneDNN would not treat it as an absolute path
+      set(DNNL_HOST_COMPILER "g++")
+      SET(DPCPP_CXX_DRIVER "icpx")
+      set(DNNL_LIB_NAME "libdnnl.a")
     else()
-      message(FATAL_ERROR "oneDNN library currently only supports GUN g++ compiler for XPU backend")
+      # Windows
+      set(DNNL_HOST_COMPILER "DEFAULT")
```

Comment:
Please check with oneDNN team what the `DEFAULT` means. Regarding Linux, it means the device compiler and host compiler are both g++. For Windows, we need to clearly understand the behavior in terms of separate compilation: MSVC for host code compilation, and ICX for device code compilation.

Reply:
Currently, `DEFAULT` is the only supported option for `ONEDNN_DPCPP_HOST_COMPILER` on Windows. I will check with oneDNN team and update.

Reply:
We may need to submit a requirement to oneDNN team to support separation compilation for Widnows. Please check with @ZhiweiYan-96 .

Reply:
`ONEDNN_DPCPP_HOST_COMPILR=DEFAULT` [follows behavior](https://oneapi-src.github.io/oneDNN/dev_guide_build_options.html#host-compiler) defined by SYCL compiler itself. According to the [compiler documentation](https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/developer-guide-reference/2024-1/fsycl-host-compiler.html#GUID-2151E257-78E0-44DC-8CB9-069F49869465) this means Intel DPC++ Compiler on Linux and Microsoft C/C++ Compiler on Windows.

+@densamoilov

Reply:
From offline discuss with @mdtoguchi, @vpirogov, the behavior of `ONEDNN_DPCPP_HOST_COMPILR=DEFAULT` on Windows is:

- The DPC++ compiler's default host compiler is Clang on Windows.
- `-fsycl-host-compiler` is `OFF` by default on Windows.
- On Windows, with `ONEDNN_DPCPP_HOST_COMPILR=DEFAULT`, `-fsycl-host-compiler` is [OFF](https://github.com/oneapi-src/oneDNN/blob/main/cmake/host_compiler.cmake#L23-L25) by default, meaning Clang is used for host compilation of oneDNN GPU library
- On Linux, with `ONEDNN_DPCPP_HOST_COMPILR=g++`, `-fsycl-host-compiler=g++` is [used](https://github.com/oneapi-src/oneDNN/blob/main/cmake/host_compiler.cmake#L103), meaning separate compilation with g++ as host compiler and DPC++ as device compiler.

The main concern of `ONEDNN_DPCPP_HOST_COMPILR=DEFAULT` on Windows is ABI compatibility, if Clang compiled oneDNN GPU library is compatible to be linked with other MSVC compiled PyTorch libraries.

Reply:
It means that 3 different compilers will be involved: MSVC, Clang, and ICPX. Is it guaranteed that Clang is always compatible with MSVC for oneDNN scope?  @mdtoguchi, @vpirogov.

Reply:
Generated binaries from `icx`, `icpx`, `icx-cl` on Windows will always be MSVC compatible unless there is an override via the `--target` option.  Use of `icx` or `icx-cl` is recommended if you require MSVC compatible command line options as well.  `icpx` will use Linux based options, but will target MSVC when run on Windows.

Reply:
Hi @mdtoguchi , let's move discussion to https://github.com/pytorch/pytorch/pull/127390. This PR had to be closed due to bad rebase.